Formula/c/cargo-clone.rb

```diff
@@ -0,0 +1,46 @@
+class CargoClone < Formula
+  desc "Cargo subcommand to fetch the source code of a Rust crate"
+  homepage "https://github.com/JanLikar/cargo-clone"
+  url "https://github.com/JanLikar/cargo-clone/archive/refs/tags/v1.2.4.tar.gz"
+  sha256 "58e86bd3440fc103572f6d8ff20ff1f99cf1e676cddb29975f204a9cb03f5b14"
+  license any_of: ["Apache-2.0", "MIT"]
+  head "https://github.com/JanLikar/cargo-clone.git", branch: "master"
+
+  depends_on "pkgconf" => :build
+  depends_on "rust" => :build
+  depends_on "rustup" => :test
+
+  depends_on "openssl@3"
+
+  uses_from_macos "curl"
+  uses_from_macos "zlib"
+
+  def install
+    # Ensure that the `openssl` crate picks up the intended library.
+    ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix
+    ENV["OPENSSL_NO_VENDOR"] = "1"
+
+    system "cargo", "install", *std_cargo_args(path: "cargo-clone")
+  end
+
+  test do
+    # Show that we can use a different toolchain than the one provided by the `rust` formula.
+    # https://github.com/Homebrew/homebrew-core/pull/134074#pullrequestreview-1484979359
+    ENV.prepend_path "PATH", Formula["rustup"].bin
+    system "rustup", "default", "beta"
+    system "rustup", "set", "profile", "minimal"
+
+    require "utils/linkage"
+
+    system bin/"cargo-clone", "clone", "git2"
```

Comment:
Should this not be something like
```suggestion
    system "cargo", "clone", "clone", "git2"
```
or else there was no point in our using `rustup` above?